### ISSUE TITLE
[BUG] Unable to sign msix due to "ST" being present in the certificate subject

### ISSUE THREAD

**Issue by danielprice-cds** [microsoft/msix-packaging#640]
*Project*
MSIX SDK (MakeAppx.exe)

**Describe the bug**
I am unable to sign a WinUI application as my code signing certificate has ST in the subject which is not permitted in the Publisher element in the AppxManifest.xml

**To Reproduce**
Create a new WinUI application and sign the app package with a code signing certificate which has a subject containing ST

**Expected behavior**
The AppxManifest.xml would allow ST in the publisher

**Comment by espresso3389** [comment]
MakeAppx.exe does not allow ST and I've finally got here and found msix-packaging has same issue... I'm using my own fork now :(

**Comment by luxzeitlos** [comment]
We're having the same problem, is there any workaround?

**Comment by florelis** [comment]
TL;DR: Use "S" instead of "ST"

The tooling uses [`CertNameToStr`](https://learn.microsoft.com/windows/win32/api/wincrypt/nf-wincrypt-certnametostra) to compute the expected publisher string from the certificate. It has some quirks mentioned in the [Remarks](https://learn.microsoft.com/windows/win32/api/wincrypt/nf-wincrypt-certnametostra#remarks), for example it uses S instead of ST, has a particular ordering for the fields, and sometimes the values need to be quoted.

I didn't confirm this, but according to past me in the [Tech Community forum](https://techcommunity.microsoft.com/discussions/msix-discussions/msix-packageing-tool--signtool-certificate-issues/224217/replies/2166455), if you set the environment variable `APPXSIP_LOG` to 3 before calling `signtool.exe`, it will tell you in the logs what publisher string it was expecting.
